t/tasks/inventory/generic/ipmi/fru_memories.t

```diff
@@ -0,0 +1,1398 @@
+            'SPEED'            => '533',
+            'TYPE'             => 'DDR2'
+        },
+        {
+            'CAPACITY'         => '2048',
+            'CAPTION'          => 'CPU1_DIMM4',
+            'DESCRIPTION'      => 'DIMM',
+            'MANUFACTURER'     => 'SK Hynix',
+            'MEMORYCORRECTION' => 'Single-bit ECC',
+            'MODEL'            => 'HYMP525P72CP4-Y5',
+            'NUMSLOTS'         => 13,
+            'SERIALNUMBER'     => '00006009',
+            'SPEED'            => '533',
+            'TYPE'             => 'DDR2'
+        },
+        {
+            'CAPACITY'         => '2048',
+            'CAPTION'          => 'CPU1_DIMM5',
+            'DESCRIPTION'      => 'DIMM',
+            'MANUFACTURER'     => 'SK Hynix',
+            'MEMORYCORRECTION' => 'Single-bit ECC',
+            'MODEL'            => 'HYMP525P72CP4-Y5',
+            'NUMSLOTS'         => 14,
+            'SERIALNUMBER'     => '00007010',
+            'SPEED'            => '533',
+            'TYPE'             => 'DDR2'
+        },
+        {
+            'CAPACITY'         => '2048',
+            'CAPTION'          => 'CPU1_DIMM6',
+            'DESCRIPTION'      => 'DIMM',
+            'MANUFACTURER'     => 'SK Hynix',
+            'MEMORYCORRECTION' => 'Single-bit ECC',
+            'MODEL'            => 'HYMP525P72CP4-Y5',
+            'NUMSLOTS'         => 15,
+            'SERIALNUMBER'     => '00001067',
+            'SPEED'            => '533',
+            'TYPE'             => 'DDR2'
+        },
+        {
+            'CAPACITY'         => '2048',
+            'CAPTION'          => 'CPU1_DIMM7',
+            'DESCRIPTION'      => 'DIMM',
+            'MANUFACTURER'     => 'SK Hynix',
+            'MEMORYCORRECTION' => 'Single-bit ECC',
+            'MODEL'            => 'HYMP525P72CP4-Y5',
+            'NUMSLOTS'         => 16,
+            'SERIALNUMBER'     => '04008004',
+            'SPEED'            => '533',
+            'TYPE'             => 'DDR2'
+        }
+    ]
+);
+
+plan tests => 3 * (scalar keys %tests) + 1;
+
+foreach my $test (keys %tests) {
+    clearFruCache();
+
+    my $dmidecode = "resources/generic/dmidecode/$test";
+    my $fru = "resources/generic/ipmitool/fru/$test";
+    my $inventory = FusionInventory::Test::Inventory->new();
+
+    lives_ok {
+        FusionInventory::Agent::Task::Inventory::Generic::Dmidecode::Memory::doInventory(
+            inventory => $inventory,
+            file      => $dmidecode
+        );
+    } "test $test: dmidecode/memory doInventory()";
+
+    lives_ok {
+        FusionInventory::Agent::Task::Inventory::Generic::Ipmi::Fru::Memory::doInventory(
+            inventory => $inventory,
+            file      => $fru
+        );
+    } "test $test: fru/memory doInventory()";
+
+    my $mem = $inventory->getSection('MEMORIES') || [];
+    my %result = map { $_->{CAPTION} => $_ } @$mem;
+    my %default = map { $_->{CAPTION} => $_ } @{$tests{$test}};
```

Comment:
Can you explain what you're doing here ? Why replacing CAPTION with a ref to the full parent hash ?This means you're not testing CAPTION content. Even here, I would better understand if `%default` would be `%expected`. Did you want to make the test not dependent on memory modules ordering ? If yes, why not just use the `bag()` as you did in `fru_controllers.t` ?

Reply:
Testing refs instead of `bag()` produces more readable errors (what and where), that's the only reason.
Hash keys i believe are still tested: the test will fail if keys differ.

Reply:
Here's what i mean:

`cmp_deeply`:
```
# Comparing hash keys of $data
# Missing: 'CPU1_DIMM70'
# Extra: 'CPU1_DIMM7'

# Compared $data->{"CPU1_DIMM7"}{"MODEL"}
#    got : 'HYMP525P72CP4-Y5'
# expect : 'HYMP525P72CP4-Y6'
```

`cmp_bag`:
```
# Comparing $data as a Bag
# Missing: 1 reference
# Extra: 1 reference

# Comparing $data as a Bag
# Missing: 1 reference
# Extra: 1 reference
```

Reply:
Okay, now, I understand. Please leave a simple comment in code to notice that.
I also must admit I read `=` in place of `=>` in the `$_->{CAPTION} => $_` expression so I misunderstood it.